android/src/main/java/com/swmansion/gesturehandler/core/GestureUtils.kt

```diff
@@ -47,4 +55,101 @@ object GestureUtils {
   }
   fun coneToDeviation(angle: Double): Double =
     cos(Math.toRadians(angle / 2.0))
+
+  fun updateStylusData(stylusData: StylusData, event: MotionEvent) {
+    // On web and iOS 0 degrees means that stylus is parallel to the surface. On android this value will be PI / 2.
+    val altitudeAngle = (PI / 2) - event.getAxisValue(MotionEvent.AXIS_TILT).toDouble()
+    val pressure = event.getPressure(0).toDouble()
+
+    var orientation = event.getOrientation(0).toDouble()
+
+    // To get azimuth angle, we need to use orientation property. Orientation value range is [-PI, PI] (https://developer.android.com/develop/ui/compose/touch-input/stylus-input/advanced-stylus-features#orientation)
+    // To shift range into [0, 2PI], we add 2PI if orientation is less than 0.
+    if (orientation < 0) {
+      orientation += 2 * PI
```

Comment:
Wouldn't that shift it to `[PI, 3PI]`?

Reply:
It's possible that I missed something, but that is why there's an if expression. I'm only shifting angles from `[-PI, 0]` - that should result in new range being `[PI, 2PI]`. Combined with second half we get range `[0, 2PI]` - a full circle.

Reply:
Also, it doesn't matter since now we use formula with modulo (mentioned in the comment below)